lib/daru/view/adapters/highcharts/display.rb

```diff
@@ -61,6 +64,47 @@ def to_html_iruby(placeholder=random_canvas_id)
       high_chart_iruby(extract_chart_class, placeholder, self)
     end
 
+    # Loads the dependent mapdata and dependent modules of the chart
+    #
+    # @param [String] to determine whether to load modules in IRuby or web
+    #   frameworks
+    # @return [void, String] loads the initial script of the modules for IRuby
+    #   notebook and returns initial script of the modules for web frameworks
+    def load_dependencies(type)
+      dep_js = extract_dependencies
+      if type == 'iruby'
+        LazyHighCharts.init_iruby(dep_js) unless dep_js.nil?
+      elsif type == 'web_frameworks'
+        dep_js.nil? ? '' : LazyHighCharts.init_script(dep_js)
+      end
+    end
+
+    # Extracts the required dependencies for the chart. User does not need
+    #   to provide any mapdata requirement explicity in the `options`.
+    #   MapData will be extracted using `options[:chart][:map]` already
+    #   provided by the user. In `modules` user needs to provide the required
+    #   modules (like tilemap in highcharts) in the form of Array. Once the
+    #   dependency is loaded on a page, there is no need to provide it again in
+    #   the `modules` option.
+    #
+    # @return [Array] the required dependencies (mapdata or modules)
+    #   to load the chart
+    # rubocop:disable Metrics/AbcSize
+    def extract_dependencies
+      dep_js = []
+      # Mapdata dependencies
+      if !options[:chart_class].nil? && options[:chart_class].capitalize == 'Map'
+        dep_js.push('mapdata/' + options[:chart][:map].to_s + '.js') if
```

Comment:
I have kept the same folder structure provided in the highcharts dependencies. It is because the value provided by the user in options[:chart][:map] will be the same as of the dependencies and by this folder structure it will be easy to retrieve the data.